packages/ui/src/box/types.ts

```diff
@@ -149,11 +151,28 @@ export type ModifiedStyledSystemProps = TypographyProps &
 export type BoxHTMLProps = React.RefAttributes<HTMLDivElement> &
   React.HTMLAttributes<HTMLDivElement>;
 
+type MarginPropNames = 'm' | 'mt' | 'mr' | 'mb' | 'ml' | 'mx' | 'my';
+type PaddingPropNames = 'p' | 'pt' | 'pr' | 'pb' | 'pl' | 'px' | 'py';
```

Comment:
there are more space props we should be using, as an example I pulled this from the styled-system type defs:
```tsx
export interface MarginProps<ThemeType extends Theme = RequiredTheme>
    extends Pick<SpaceProps<ThemeType>,
    | 'm'
    | 'margin'
    | 'mt'
    | 'marginTop'
    | 'mb'
    | 'marginBottom'
    | 'ml'
    | 'marginLeft'
    | 'mr'
    | 'marginRight'
    | 'my'
    | 'marginY'
    | 'mx'
    | 'marginX'> {
}
```

Reply:
This is the entire space props:
```tsx
export interface SpaceProps<ThemeType extends Theme = RequiredTheme, TVal = ThemeValue<'space', ThemeType>> {
    /** Margin on top, left, bottom and right */
    m?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on top, left, bottom and right */
    margin?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on top */
    mt?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on top */
    marginTop?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on right */
    mr?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on right */
    marginRight?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on bottom */
    mb?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on bottom */
    marginBottom?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on left */
    ml?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on left */
    marginLeft?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on left and right */
    mx?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on left and right */
    marginX?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on top and bottom */
    my?: ResponsiveValue<TVal, ThemeType>;
    /** Margin on top and bottom */
    marginY?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on top, left, bottom and right */
    p?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on top, left, bottom and right */
    padding?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on top */
    pt?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on top */
    paddingTop?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on right */
    pr?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on right */
    paddingRight?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on bottom */
    pb?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on bottom */
    paddingBottom?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on left */
    pl?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on left */
    paddingLeft?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on left and right */
    px?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on left and right */
    paddingX?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on top and bottom */
    py?: ResponsiveValue<TVal, ThemeType>;
    /** Padding on top and bottom */
    paddingY?: ResponsiveValue<TVal, ThemeType>;
}
```

Reply:
Good feedback, can do

```
type MarginPropNames = keyof StyledSystem.MarginProps;
type PaddingPropNames = keyof StyledSystem.PaddingProps;
```